scientific_library/tvb/simulator/monitors.py

```diff
@@ -856,6 +857,10 @@ def compute_hrf(self):
 
     def config_for_sim(self, simulator):
         super(Bold, self).config_for_sim(simulator)
+
+        if self.period > simulator.simulation_length:
```

Comment:
I fear this might be too early as a check for the library mode. See the examples there: config happens early, the the simulator __call__ loop only gets the simulation length. Other examples configure everything, do a small loop (initial results are ignored), then continue the simulation for a longer length.